core/src/main/java/org/elasticsearch/index/store/StoreFileMetaData.java

```diff
@@ -79,6 +69,27 @@ public StoreFileMetaData(String name, long length, String checksum, Version writ
         this.hash = hash == null ? new BytesRef() : hash;
     }
 
+    /**
+     * Read from a stream.
+     */
+    public StoreFileMetaData(StreamInput in) throws IOException {
+        name = in.readString();
+        length = in.readVLong();
+        checksum = in.readString();
+        String versionString = in.readString();
+        assert versionString != null;
+        writtenBy = Lucene.parseVersionLenient(versionString, FIRST_LUCENE_CHECKSUM_VERSION);
```

Comment:
This is the source of the confusion. Why is this like this? Why isn't it just a straight read of the string off the stream?